shell/common/shell.cc

```diff
@@ -610,6 +611,18 @@ void Shell::OnPlatformViewSetViewportMetrics(const ViewportMetrics& metrics) {
   FML_DCHECK(is_setup_);
   FML_DCHECK(task_runners_.GetPlatformTaskRunner()->RunsTasksOnCurrentThread());
 
+  // Only set this if the user has _not_ requested a specific value.
+  if (raster_cache_max_bytes_user_value_ == -1) {
+    // This is the formula Android uses.
```

Comment:
Can we add a link to the Android formula? I'm curious how it gets the magic number 12.

Reply:
See https://android.googlesource.com/platform/frameworks/base/+/master/libs/hwui/renderthread/CacheManager.cpp#41

Reply:
Added link to comment